packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -534,6 +541,91 @@ export class CdkToolkit {
     return stacks;
   }
 
+  private stackHasNestedStacks(rootStack: cxapi.CloudFormationStackArtifact) {
+    for (const resourceName in rootStack.template.Resources) {
+      if (rootStack.template.Resources[resourceName].Type === 'AWS::CloudFormation::Stack') {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private async findPhysicalNameForNestedStack(logicalId: string, parentStackName: string | undefined, sdk: ISDK): Promise<string | undefined> {
+    if (!parentStackName) {
+      return undefined;
+    }
+    const stackResources = await this.listNestedStackResources(parentStackName, sdk);
+    return stackResources.find(sr => sr.LogicalResourceId === logicalId)?.PhysicalResourceId;
+  }
+
+  private async listNestedStackResources(parentStackName: string, sdk: ISDK): Promise<CloudFormation.StackResourceSummary[]> {
+    const ret = new Array<CloudFormation.StackResourceSummary>();
+    let nextToken: string | undefined;
+    do {
+      const stackResourcesResponse = await sdk.cloudFormation().listStackResources({
+        StackName: parentStackName,
+        NextToken: nextToken,
+      }).promise();
+      ret.push(...(stackResourcesResponse.StackResourceSummaries ?? []));
+      nextToken = stackResourcesResponse.NextToken;
+    } while (nextToken);
+    return ret;
+  }
+
+  private async getNestedStackTemplates(
+    root: cxapi.CloudFormationStackArtifact, nestedTemplateAssetPath: string, nestedStackLogicalId: string,
```

Comment:
This name (`root`) is a little too terse... this should be something like `rootStackArtifact`, or something.

Same comment in the remaining places in this PR.